### PR TITLE
feat: map the user type to the ui on conversation

### PR DESCRIPTION
**Pull request by trOnk12**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

- Kalium exposes now a user type on conversation details, which we can map now into the ui 

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-606/1/) **failed**.

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-606/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3449993)(3449993)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3449993)(3449993)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewModel.kt [hunk 1]
```diff
@@ -447,6 +449,18 @@ class ConversationViewModel @Inject constructor(
         }
     }
 
+    private fun mapUserType(userType: UserType): Membership {
+        return when (userType) {
+            UserType.GUEST -> Membership.Guest
+            UserType.FEDERATED -> Membership.Federated
+            UserType.EXTERNAL -> Membership.External
+            UserType.INTERNAL -> Membership.None
+            else -> {
```
Comment (vitorhugods) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewModel.kt:458]: Isn't `UserType` an `enum` ?

We don't need the `else` here in order to compile the code.

With the `else`, if we add new values in the future, this won't fail in compile time and the exception will be thrown unnecessarily.
I argue that handling new values in build time is better than having these silent errors crashing the app.

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewModel.kt [hunk 2]
```diff
@@ -447,6 +449,18 @@ class ConversationViewModel @Inject constructor(
...
+                throw IllegalStateException("Unknown UserType")
+            }
+        }
+    }
+
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewModel.kt:463]: Couldn't we have this in a `ConversationMapper`?

**Review by alexandreferris — COMMENTED** [review]
Looking great! I would just suggest having the mapping function maybe in a Mapper class in case we might need to use it in the future so we avoid duplication.

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewModel.kt (hunk 2 above)
Comment (trOnk12) [reply app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewModel.kt:463]: membership is kind of a. UI model do not think Kalium should know about it ? I added a UserTypeMapper to the AR 

I think in general it is good idea to use mapper classes inside the AR :)

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-606/4/) **failed**.

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-606/5/) **failed**.

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-606/6/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3451183)(3451183)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3451183)(3451183)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [7](https://10.10.124.134/job/wire-android-reloaded/job/PR-606/7/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3451505)(3451505)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3451505)(3451505)-dev-release.apk)

**Review by vitorhugods — APPROVED** [review]
✅🚀 Nicely done!

**Pull request merged**